src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -963,4 +989,38 @@ bool EventPipe::IsLockOwnedByCurrentThread()
 }
 #endif
 
+// This method will block runtime bring-up IFF DOTNET_DiagnosticsMonitorAddress != nullptr and DOTNET_DiagnosticsMonitorStopOnStart!=0 (it's default state)
+// The s_ResumeRuntimeStartupEvent event will be signaled when the Diagnostics Monitor uses the ResumeRuntime Diagnostics IPC Command
+void EventPipe::PauseForTracingAgent()
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        PRECONDITION(s_ResumeRuntimeStartupEvent->IsValid());
+    }
+    CONTRACTL_END;
+
+    CLRConfigStringHolder pDotnetDiagnosticsMonitorAddress = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_DOTNET_DiagnosticsMonitorAddress);
+    DWORD dwDotnetDiagnosticsMonitorStopOnStart = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_DOTNET_DiagnosticsMonitorStopOnStart);
+    if (pDotnetDiagnosticsMonitorAddress != nullptr && dwDotnetDiagnosticsMonitorStopOnStart != 0)
```

Comment:
Can you early out after reading the first config switch? We do not need to read the second config switch once we determine that the first one is null.